oak/proto/permissions.proto

```diff
@@ -0,0 +1,59 @@
+//
+// Copyright 2020 The Project Oak Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+syntax = "proto3";
+
+package oak.permissions;
+
+// A PermissionsConfiguration provides a declarative description of the features that are permitted
+// for an Oak application.
+//
+// The permissions are applied dynamically at runtime, when creating the Nodes.
+message PermissionsConfiguration {
+  // Whether to enable the use of a GrpcServerNode as the front-door to the application.
```

Comment:
Is there a difference between using a gRPC server node as front-door vs non front-door? Maybe it would be simpler to just call the various options like the corresponding pseudo-nodes that they enable?

Reply:
I think there will be only one GrpcServerNode in an Oak application. But there could potentially be several HttpServer nodes, for example for application monitoring. Is that right? 

My intention with `grpc_front_door` and `http_front_door` is to allow specifying how the application is exposed to the outside world. In case of the HttpServerNodes, I think we would want to separate the config for the front-door, and the monitoring server node (currently no flag for it here BTW).

Or maybe we just need to allowlist the egress traffic? Again in the case of HTTP servers, we probably need separate allowlist for the front-door and monitoring servers.

Reply:
As discussed earlier today, it is now a flag for allowing the creation of any grpc server pseudo-node. 
Also as discussed, there is no need to have any limitations on ingress traffic, as it is too difficult to maintain, and the functionality is generally covered by firewalls.